### ISSUE TITLE
y_text - not detecting GameText colors

### ISSUE THREAD

**Issue by Vasily-X** [pawn-lang/YSI-Includes#146]
I don't know why it doesn't detect GameText colors when they're used before the first word of the message. This happens only for client messages.
```
[my_section]
MY_MESSAGE = ~r~One {GOLD}Two ~r~Three
```
"Two" and "Three" are printed correctly, but "One" is black (by default) or the color I set in DATA_LANG.yml.

AFAIK the colour attribute from DATA_LANG.yml doesn't support GameText colors either.

Thank you.

**Comment by Vasily-X** [comment]
Well, I guess it doesn't care at all as it happens to the first word only, fixable through the color attribute from the .yml file. Maybe that's how it's supposed to work after all.

**Comment by Y-Less** [comment]
Not sure what you think doesn't care. This does sound like a bug, and should really remain open, even if it hasn't been got to yet.

**Comment by Vasily-X** [comment]
@Y-Less I thought it doesn't make sense after all, I mean, if the color attribute works okay why would I add extra characters to the string? Anyway, you're right, if this is a bug it should be fixed.

I've been trying to debug and track the error, didn't succeed unfortunately.

Gonna try with #147

**Comment by Y-Less** [comment]
I've been neglecting y_text bugs for a long time because I started trying to rewrite it in a more modular testable fashion.  However, that's just not working - the more I do, the more I realise the old version did and the more corner-cases it handled quite well.  The code might be messy, but it worked well for the most part other than a few issues with colours.  I'm just going to abandon efforts to rewrite it (since a ground-up rewrite is rarely the best solution anyway) and work on fixing the original.  I've also got some ideas for performance improvements but they're just modifications to existing code, not starting from scratch.

Point being, I'll try get to these now.
